packages/react-composites/src/composites/common/ControlBar/CommonCallControlBar.tsx

```diff
@@ -109,493 +110,509 @@ const inferCommonCallControlOptions = (
   return options;
 };
 
+type CommonCallControlBarMergedProps = CommonCallControlBarProps & ContainerRectProps;
+
 /**
  * @private
  */
-export const CommonCallControlBar = (props: CommonCallControlBarProps & ContainerRectProps): JSX.Element => {
-  const theme = useTheme();
-  const rtl = theme.rtl;
-
-  const controlBarContainerRef = useRef<HTMLHeadingElement>(null);
-  const sidepaneControlsRef = useRef<HTMLHeadingElement>(null);
-  const controlBarSizeRef = useRef<HTMLHeadingElement>(null);
-
-  const [controlBarButtonsWidth, setControlBarButtonsWidth] = useState(0);
-  const [panelsButtonsWidth, setPanelsButtonsWidth] = useState(0);
-  const [controlBarContainerWidth, setControlBarContainerWidth] = useState(0);
-
-  const [totalButtonsWidth, setTotalButtonsWidth] = useState(0);
-  const [isOutOfSpace, setIsOutOfSpace] = useState(false);
-
-  const callWithChatStrings = useCallWithChatCompositeStrings();
-  const options = inferCommonCallControlOptions(props.mobileView, props.callControls);
-
-  const [showCaptionsSettingsModal, setShowCaptionsSettingsModal] = useState(false);
-
-  // If the hangup capability is not present, we default to true
-  const isHangUpForEveryoneAllowed =
-    useSelector((state) => state.call?.capabilitiesFeature?.capabilities.hangUpForEveryOne.isPresent) ?? true;
-  const isTeams = useSelector(getIsTeamsCall);
-
-  /* @conditional-compile-remove(breakout-rooms) */
-  const assignedBreakoutRoom = useSelector(getAssignedBreakoutRoom);
-  /* @conditional-compile-remove(breakout-rooms) */
-  const breakoutRoomSettings = useSelector(getBreakoutRoomSettings);
-
-  const handleResize = useCallback((): void => {
-    setControlBarButtonsWidth(controlBarContainerRef.current ? controlBarContainerRef.current.offsetWidth : 0);
-    setPanelsButtonsWidth(sidepaneControlsRef.current ? sidepaneControlsRef.current.offsetWidth : 0);
-    setControlBarContainerWidth(controlBarSizeRef.current ? controlBarSizeRef.current.offsetWidth : 0);
-  }, []);
-
-  // on load set inital width
-  useEffect(() => {
-    setControlBarButtonsWidth(controlBarContainerRef.current ? controlBarContainerRef.current.offsetWidth : 0);
-    setPanelsButtonsWidth(sidepaneControlsRef.current ? sidepaneControlsRef.current.offsetWidth : 0);
-    setControlBarContainerWidth(controlBarSizeRef.current ? controlBarSizeRef.current.offsetWidth : 0);
-  }, []);
-
-  // get the current width of control bar buttons and panel control buttons when browser size change
-  useEffect(() => {
-    window.addEventListener('resize', handleResize);
-    return () => window.removeEventListener('resize', handleResize);
-  }, [handleResize]);
-
-  /* when size change, reset total buttons width and compare with the control bar container width
+export const CommonCallControlBar = forwardRef<FocusableElement, CommonCallControlBarMergedProps>(
```

Comment:
Annoying to review this as it's mostly just identation change. But to spell out the changes a bit:
- Added`forwardRef` here to allow creating imperative handle
- Created imperativeHandle that exposes a focus method that resets the `key` of the hidden element to force remount
   ```
    // Create an imperatively callable focus function to allow the parent component to focus the control bar any time.
    const [hiddenFocusStartElementKey, setHiddenFocusStartElementKey] = useState(0);
    useImperativeHandle<FocusableElement, FocusableElement>(commonCallControlBarRef, () => ({
      focus: () => {
        setHiddenFocusStartElementKey((prevKey) => prevKey + 1);
      }
    }));
   ```
- Apply key to hidden element: `<HiddenFocusStartPoint key={hiddenFocusStartElementKey} />`

Reply:
Kicks focus back to the start of the control bar. For people pane back to people button this wouldn't work as you'd want focus to go to a specific place (the people pane) instead of the start of the control bar